### PR TITLE
Make trademark checks less strict

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/14945

### CODE REVIEW

**Comment by willdurand** [comment]
Test failures: ES tests

**Comment by willdurand** [comment]
Ha wait, looks like we still need some checks.

**Comment by willdurand** [comment]
Patch updated. cc @wagnerand

File: src/olympia/addons/utils.py [hunk 1]
```diff
@@ -28,7 +28,12 @@ def _check(name):
         name = normalize_string(name, strip_punctuation=True).lower()
 
         for symbol in amo.MOZILLA_TRADEMARK_SYMBOLS:
-            if symbol in name:
+            violates_trademark = (
+                name.count(symbol) > 1 or (
+                    name.count(symbol) >= 1 and not
+                    name.endswith(' for {}'.format(symbol))))
```
Comment (diox) [inline src/olympia/addons/utils.py:34]: I'm confused, this doesn't *remove* the trademark restriction code, but patches it ?

Comment (willdurand) [reply src/olympia/addons/utils.py:34]: Yeah, looks like we need to remove some restrictions but not everything. I reverted the previous patch that added stricter checks, and removed the waffle switch that was used to enable/disable the stricter checks.

**Pull request merged**